src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -7244,7 +7343,7 @@ void Player_Action_80840450(Player* this, PlayState* play) {
 
     func_8083721C(this);
 
-    if (!func_80837348(play, this, D_808543E0, 1)) {
+    if (!Player_TryActionChangeList(play, this, sActionChangeList1, true)) {
```

Comment:
Something I noticed is that most uses of `Player_TryActionChangeList` can be replaced with an early return:

```c
if (Player_TryActionChangeList(play, this, sActionChangeList1, true)) {
    return;
}
...
```

Which I find helps the readability, indentation, and logic of this function succeeding in changing the action. Just my personal preference.

Changing it here though would make some bigger diffs due to indentation, so it should probably be done separate if we wanted to make these all early returns

Reply:
Definitely something to explore. Would agree on waiting till a future PR.

To me, having everything else that is allowed to continue running be indented makes it more clear to me that there is a "possibility for interruption", if that makes sense. I feel like with an early return itd be easier to miss that control flow detail. Whereas if I see its indented i make sure to scroll up and see what if condition is blocking it.

But im not totally against the idea and will def see how it would play out in practice.

Reply:
MM currently has early returns in most places for this function, so you could see how it could eventually look like. Personally I think it looks better with the early returns.